spiffworkflow-backend/src/spiffworkflow_backend/scripts/create_unique_kkv_top_level_key.py

```diff
@@ -19,9 +21,23 @@ def get_description(self) -> str:
         )
 
     def run(self, script_attributes_context: ScriptAttributesContext, *args: Any, **kwargs: Any) -> Any:
-        top_level_key_prefix = args[0]
+        identifier = args[0]
+        top_level_key_prefix = args[1]
+        spiff_task = script_attributes_context.task
+        location: str | None = None
 
-        model = KKVDataStoreModel(top_level_key="", secondary_key="", value={})
+        if identifier is not None and spiff_task is not None:
+            location = DataStoreCRUD.data_store_location_for_task(KKVDataStoreModel, spiff_task, identifier)
+
+        instance_model: KKVDataStoreModel | None = None
+
+        if location is not None:
+            instance_model = db.session.query(KKVDataStoreModel).filter_by(identifier=identifier, location=location).first()
+
+        if instance_model is None:
+            raise Exception(f"Could not find KKV data store with the identifier '{identifier}'")
+
+        model = KKVDataStoreEntryModel(instance_id=instance_model.id, top_level_key="", secondary_key="", value={})
```

Comment:
Right now the database is the one shared state that we have. When running a workflow you could have a script task that needs a unique id across the system, then starts building up a kkv data store, then saves it. I do agree this could introduce unnecessary db operations - but right now this feature is very infrequently used. Will take a note to consider other options for generating the key.